### ISSUE TITLE
Possible false positives using Mockito

### ISSUE THREAD

**Issue by ZOlbrys** [uber/NullAway#280]
Hi again! Another potential issue that I wanted to run by the team...

Consider the following code which runs a unit test on some code, using `mockito` to create a mock object for use in the test:

```
@RunWith(MockitoJUnitRunner.class)
public class ExampleInstrumentedTest {
    @Mock
    SampleDAO mockDAO;

    @Test
    public void testMockitoNullWarnings() {
        final CustomValue mockCustomValue = mock(CustomValue.class);
        final int mockId = 1;

        when(mockCustomValue.getName()).thenReturn("Zach");
        when(mockDAO.getValueWithId(mockId)).thenReturn(mockCustomValue);
        final SampleClass sampleClass = new SampleClass(mockDAO);

        sampleClass.doSomething();

        verify(mockDAO, times(1)).getValueWithId(mockId);
    }
}
```
I see the following reports when running my unit tests:

```
> Task :app:compileDebugAndroidTestJavaWithJavac
/Users/zach/Developer/Code/ErrorProneExample/app/src/androidTest/java/com/example/errorproneexample/ExampleInstrumentedTest.java:21: warning: [NullAway] @NonNull field mockDAO not initialized
    SampleDAO mockDAO;
              ^
    (see http://t.uber.com/nullaway )
1 warning
```
Is this a false positive? Should this be safely ignored? To my knowledge `@Mock` will always return a nonnull object, but I can't find documentation proving this.  I'm not sure if this is a NullAway issue or if it's a Mockito issue, or not an issue at all :P.

I am using nullaway 0.6.4 setup via errorprone in an Android project.  A sample of the issue is available here: https://github.com/ZOlbrys/ErrorProneExample with my latest commits.

**Comment by lazaroclapp** [comment]
There is already support for this case, actually, you can pass annotations to `-XepOpt:NullAway: ExcludedFieldAnnotations=...` and that will cause initialization checking to ignore fields annotated with those.

We use it for `dagger`, and that is supported out of the box. We don't have hardcoded support for `mockito` (perhaps we should? @msridhar ) but you can easily configure NullAway to understand which annotations cause external initialization.

**Comment by msridhar** [comment]
Yeah we could think about supporting mockito out of the box but I’d want to think about that more. In the meantime @lazaroclapp’s solution is the way to go

**Comment by ZOlbrys** [comment]
That works very well.  Thanks for the information.  I've confirmed that this fixes the warnings in my sample (and real) projects.  🥇

**Comment by msridhar** [comment]
Great!  I'm going to close for now.  If we get more reports in the future we can think about built-in Mockito support
